### PR TITLE
tests: Fix mypy errors and run mypy during PR build

### PR DESCRIPTION
**Pull request by bkeryan**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nidaqmx-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Fix mypy errors in tests. For details, see commits.

Update PR build workflow to install extras and run mypy (tests only, for now).

### Why should this Pull Request be merged?

Fix mypy errors that I found when I added the nidaqmx-python test fixtures to an internal test component.

Ooch towards being able to run mypy on the entire component.

### What testing has been done?

Ran `poetry run mypy tests` and `poetry run pytest -v`

### CODE REVIEW

File: tests/unit/test_grpc_time.py [hunk 1]
```diff
@@ -10,14 +10,10 @@
...
 
 try:
-    from google.protobuf.timestamp_pb2 import Timestamp as GrpcTimestamp
-
     import nidaqmx._grpc_time as grpc_time
     import nidaqmx._stubs.nidaqmx_pb2 as nidaqmx_pb2
 except ImportError:
-    GrpcTimestamp = None
-    grpc_time = None
-    nidaqmx_pb2 = None
```
Comment (bkeryan) [inline tests/unit/test_grpc_time.py:20]: We only need to assign things to None when we want to check them at run time.

File: tests/component/test_stream_readers_di.py [hunk 2]
```diff
@@ -179,12 +179,12 @@ def _bool_array_to_int(bool_array: numpy.typing.NDArray[numpy.bool_]) -> int:
     return result
 
 
-_DType = TypeVar("_DType", bound=numpy.typing.DTypeLike)
+_D = TypeVar("_D", bound=numpy.generic, covariant=True)
```
Comment (bkeryan) [inline tests/component/test_stream_readers_di.py:182]: @zhindes It turns out numpy.generic was right after all, but we want a type var to make the return type preserve the parameter type.

This also implicitly relies on the fact that the DAQmx read methods are untyped (Any).

File: tests/_grpc_utils.py [hunk 3]
```diff
@@ -48,22 +50,23 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self._stdout_thread.join()
 
     def _get_grpc_server_exe(self):
-        if os.name != "nt":
-            pytest.skip("Only supported on Windows")
-        import winreg
+        if sys.platform == "win32":
+            import winreg
```
Comment (bkeryan) [inline tests/_grpc_utils.py:54]: mypy recognizes `sys.platform` but not necessarily `os.name`

Also this relied on mypy knowing that pytest.skip never returns, but apparently it does know that: 
https://github.com/pytest-dev/pytest/blob/8a410d0ba60bca7073bf18a1503c59a3b5445780/src/_pytest/outcomes.py#L120

Comment (bkeryan) [reply tests/_grpc_utils.py:54]: ...which explains all the "unreachable code" warnings for return statements after a pytest.skip

**Review by zhindes — APPROVED** [review]

**Review by maxxboehme — APPROVED** [review]

**Pull request merged**